test/widgets/message_list_test.dart

```diff
@@ -981,4 +983,222 @@ void main() {
       });
     });
   });
+
+  group('message list page for DMs with deactivated users', () {
```

Comment:
Does the new logic interact with the message list? I don't think it does — it seems like it's all contained in the compose box.

In that case these tests should go in `test/widgets/compose_box_test.dart` instead, and should be able to use `prepareComposeBox` there (or something like it) without setting up a MessageListPage.

In general when adding code in a file `lib/foo/bar.dart`, we try to put the tests for that code in the corresponding file `test/foo/bar_test.dart`.